docs/zkapps/tutorials/12-cross-contract-calls.mdx

```diff
@@ -0,0 +1,209 @@
+
+1. First, open the [Composability.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/12-cross-contract-calls/src/Composability.ts) example file.
+	
+1. Copy the file's entire contents into your project `src/Composability.ts` file.
+
+### Imports and Incrementer smart contract 
+
+First, bring in imports and set up the first smart contract `Incrementer`.
+
+```typescript
+import {
+  Field,
+  method,
+  Mina,
+  AccountUpdate,
+  PrivateKey,
+  SmartContract,
+  state,
+  State,
+} from 'o1js';
+
+class Incrementer extends SmartContract {
+  @method increment(x: Field): Field {
+    return x.add(1);
+  }
+}
+
+```
+This Incrementer contract adds `1` to the `Field` argument, which is passed.
+
+### Adder smart contract 
+
+Now bring in the second contract `Adder` that returns the addition of two numbers and adds `1` to the result.
+The addition of `1` to the result is outsourced to the `Incrementer` smart contract by creating a new object by passing its address.
+
+```typescript
+class Adder extends SmartContract {
+  @method addPlus1(x: Field, y: Field): Field {
+    let sum = x.add(y);
+    let incrementer = new Incrementer(incrementerAddress);
+    return incrementer.increment(sum);
+  }
+}
+
+```
+
+### Caller smart contract 
+
+The final smart contract `Caller` calls the `addPlus1()` method of the `Adder` smart contract and emits the stored result that is returned.
+
+```typescript
+class Caller extends SmartContract {
+  @state(Field) sum = State<Field>();
+  events = { sum: Field };
+
+  @method callAddAndEmit(x: Field, y: Field) {
+    let adder = new Adder(adderAddress);
+    let sum = adder.addPlus1(x, y);
+    this.emitEvent('sum', sum);
+    this.sum.set(sum);
+  }
+}
+
+```
+
+The code to interact with the smart contract:
+
+```typescript
+const doProofs = true;
+
+let Local = Mina.LocalBlockchain({ proofsEnabled: doProofs });
+Mina.setActiveInstance(Local);
+
+let feePayerKey = Local.testAccounts[0].privateKey;
+let feePayer = Local.testAccounts[0].publicKey;
+
+let incrementerKey = PrivateKey.random();
+let incrementerAddress = incrementerKey.toPublicKey();
+
+let adderKey = PrivateKey.random();
+let adderAddress = adderKey.toPublicKey();
+
+let zkappKey = PrivateKey.random();
+let zkappAddress = zkappKey.toPublicKey();
+
+let zkapp = new Caller(zkappAddress);
+let adderZkapp = new Adder(adderAddress);
+let incrementerZkapp = new Incrementer(incrementerAddress);
+
+if (doProofs) {
+  console.log('compile (incrementer)');
+  await Incrementer.compile();
+  console.log('compile (adder)');
+  await Adder.compile();
+  console.log('compile (caller)');
+  await Caller.compile();
+}
+
+console.log('deploy');
+let tx = await Mina.transaction(feePayer, () => {
+  AccountUpdate.fundNewAccount(feePayer, 3);
+  zkapp.deploy();
+  adderZkapp.deploy();
+  incrementerZkapp.deploy();
+});
+await tx.sign([feePayerKey, zkappKey, adderKey, incrementerKey]).send();
```

Comment:
@LuffySama-Dev Can you also leave a comment about what is happening here?

Reply:
Hi @iregina ,
Yes sure. I will add a comment.

Thank You ☺️

Reply:
I like your question about comments @LuffySama-Dev  
Helpful comments in the code is the pathway to sharing knowledge 

Please add descriptive comments

Reply:
Hi @barriebyron , @iregina ,

I have committed the changes, can you please help to check and let me know if any changes are required ?
- Also, I am planning on adding a flow diagram of how call is going from one contract to other an so.
What do you think ?

Thank You 😃